CMakeLists.txt

```diff
@@ -510,6 +513,25 @@ if(WITH_CORE)
     set(HAVE_3D TRUE)  # used in qgsconfig.h
   endif()
 
+  set (WITH_QTWEBENGINE FALSE CACHE BOOL "Enable QtWebEngine support")
+  if (WITH_QTWEBENGINE)
+    if(WITH_3D)
+      message(FATAL_ERROR "QtWebEngine support cannot be enabled if QGIS 3D support is enabled")
+    endif()
+    if(WITH_QTWEBKIT)
+      message(FATAL_ERROR "QtWebEngine support cannot be enabled if Qt WebKit support is enabled")
+    endif()
+
+    find_package(${QT_VERSION_BASE} COMPONENTS WebEngineCore WebEngineWidgets REQUIRED)
+
+    add_definitions(-DWITH_QTWEBENGINE)
```

Comment:
Should we start moving these to the more modern target based system?
I.e. this should be in core/CMakeLists.txt next to the link library instructions.

```cmake
target_compile_definitions(qgis_core PUBLIC WITH_QTWEBENGINE)
```

This could also be done in a separate follow up which is applied on the complete code base, if there are no objections.

Reply:
Turns out it was redundant anyway. I've removed it now.